coord/src/rmr/endpoint.c

```diff
@@ -13,6 +13,7 @@
 int MREndpoint_Parse(const char *addr, MREndpoint *ep) {
 
   ep->host = NULL;
+  ep->unixSock = NULL;
```

Comment:
This section of the function actually what zeroes up the struct. I agree that it’s a bit strange but I guess it was a bit useful because the shard struct is static.
I can zero it up outside and remove this part entirely (or zero it up completely and not by field). The thing is that if these values won’t be NULLs when we start parsing, we will attempt to free them if we get a parsing error. I think this is safer to do in general